go/arrow/array/numericbuilder.gen.go

```diff
@@ -89,10 +89,12 @@ func (b *Int64Builder) AppendValues(v []int64, valid []bool) {
 		panic("len(v) != len(valid) && len(valid) != 0")
 	}
 
-	b.Reserve(len(v))
-	if len(v) > 0 {
-		arrow.Int64Traits.Copy(b.rawData[b.length:], v)
+	if len(v) == 0 {
```

Comment:
@stuartcarnie: could I pick your brain on this?
it seems to me this is a valid change, but I'd like to know whether I didn't overlook anything...
(all tests pass, but...)